File: Demo/Sources/ViewControllers/ScrollView/ScrollViewController.m
//
//  Globus
//

#import "ScrollViewController.h"

@interface ScrollViewController ()

@property(nonatomic, weak) IBOutlet GLBListField* listField;
@property(nonatomic, weak) IBOutlet GLBScrollView* scrollView;

@end

@implementation ScrollViewController

#pragma mark - UIViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _listField.items = @[
        [GLBListFieldItem itemWithTitle:@"Stretch" value:@(GLBScrollViewDirectionStretch)],
        [GLBListFieldItem itemWithTitle:@"Horizontal" value:@(GLBScrollViewDirectionHorizontal)],
        [GLBListFieldItem itemWithTitle:@"Vertical" value:@(GLBScrollViewDirectionVertical)]
    ];
    _scrollView.direction = GLBScrollViewDirectionStretch;
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

- (IBAction)changedSelectedItem:(id)sender {
    _scrollView.direction = [_listField.selectedItem.value integerValue];
}

@end

File: Demo/Sources/ViewControllers/ScrollView/ScrollViewController.swift
//
//  Globus
//

import Globus

class ScrollViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet fileprivate weak var listField: GLBListField!
    @IBOutlet fileprivate weak var scrollView: GLBScrollView!
    
    // MARK - UIViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.listField.items = [
            GLBListFieldItem(title: "Stretch", value:GLBScrollViewDirection.stretch),
            GLBListFieldItem(title: "Horizontal", value:GLBScrollViewDirection.horizontal),
            GLBListFieldItem(title: "Vertical", value:GLBScrollViewDirection.vertical)
        ]
        self.scrollView.direction = .stretch
    }
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    @IBAction internal func changedSelectedItem(_ sender: Any) {
        if let direction = self.listField.selectedItem?.value as? GLBScrollViewDirection {
            self.scrollView.direction = direction
        }
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "ScrollViewController-Swift"
    }
    
}
